CoreLayersView/test/unit/src/au/gov/asd/tac/constellation/views/layers/utilities/LayersDualGraphSyncNGTest.java

```diff
@@ -0,0 +1,318 @@
+
+        graph.setObjectValue(colorV, vxId1, vx1Color);
+        graph.setObjectValue(colorV, vxId2, vx2Color);
+        graph.setObjectValue(colorV, vxId3, vx3Color);
+//        graph.setIntValue(layerMaskV, vxId1, 1);
+//        graph.setFloatValue(layerVisibilityV, vxId1, 1.0f);
+//        graph.setBooleanValue(selectedV, vxId1, false);
+//
+//        vxId2 = graph.addVertex();
+//        graph.setIntValue(layerMaskV, vxId2, 1);
+//        graph.setFloatValue(layerVisibilityV, vxId2, 1.0f);
+//        graph.setBooleanValue(selectedV, vxId2, false);
+    }
+
+    @Test
+    public void dynamicLayerChangeTest() throws InterruptedException, PluginException {
+        setupGraph();
+
+        // Check Vertex set correctly
+        assertTrue(1 == graph.getIntValue(layerMaskV, vxId1));
+        assertTrue(1.0f == graph.getFloatValue(layerVisibilityV, vxId1));
+
+        assertTrue(1 == graph.getIntValue(layerMaskV, vxId2));
+        assertTrue(1.0f == graph.getFloatValue(layerVisibilityV, vxId2));
+
+        assertTrue(1 == graph.getIntValue(layerMaskV, vxId3));
+        assertTrue(1.0f == graph.getFloatValue(layerVisibilityV, vxId3));
+        queries.clear();
+        queries.add("Default");
+        queries.add("color == " + vx1Color.toString());
+
+        LayersConcept.GraphAttribute.LAYER_QUERIES.ensure(graph);
+        LayersConcept.GraphAttribute.LAYER_PREFERENCES.ensure(graph);
+
+        PluginExecution.withPlugin(new UpdateGraphQueriesPlugin(queries)).executeNow(graph);
+        PluginExecution.withPlugin(new UpdateGraphBitmaskPlugin(0b10)).executeNow(graph);
+
+        // Check Vertex set correctly
+        assertTrue(3 == graph.getIntValue(layerMaskV, vxId1));
+        assertTrue(1.0f == graph.getFloatValue(layerVisibilityV, vxId1));
+        assertTrue(vx1Color.toString().equals(graph.getObjectValue(colorV, vxId1).toString()));
+
+        assertTrue(1 == graph.getIntValue(layerMaskV, vxId2));
+        assertTrue(0.0f == graph.getFloatValue(layerVisibilityV, vxId2));
+        assertTrue(vx2Color.toString().equals(graph.getObjectValue(colorV, vxId2).toString()));
+
+        assertTrue(1 == graph.getIntValue(layerMaskV, vxId3));
+        assertTrue(0.0f == graph.getFloatValue(layerVisibilityV, vxId3));
+        assertTrue(vx3Color.toString().equals(graph.getObjectValue(colorV, vxId3).toString()));
+
+        // change an attribute to trigger switching of graph object within dual graph
+        PluginExecution.withPlugin(new SimpleEditPlugin("Test: change attribute value") {
+            @Override
+            public void edit(final GraphWriteMethods graph, final PluginInteraction interaction, final PluginParameters parameters) throws InterruptedException {
+                final int maxTransactionsId = VisualConcept.GraphAttribute.MAX_TRANSACTIONS.ensure(graph);
+                graph.setIntValue(maxTransactionsId, 0, 9);
+            }
+        }).executeNow(GraphManager.getDefault().getActiveGraph());
```

Comment:
@aldebaran30701 this is similar to the problem I am presently looking at. In a unit test `getActiveGraph()` will be `null` so you will find that this plugin isn't executed. In the `edit()` if you print something out I think you will find that it isn't appearing.

The fix would be to pass in `graph`.

Reply:
@aldebaran30701 I updated it to use `graph` and it passed - so executing `graph.setIntValue(maxTransactionsId, 0, 9);` had no net effect. Do you think a revisit of the unit test or layers code is required?

Reply:
That would explain why the graph out of sync issue was not reproducible through the unit tests. I assumed it was a product of running it on a single thread.

Reply:
So I have also had a play around with this, and it seems that because `getActiveGraph()` returns null, and graph is the same instance of `StoreGraph`, I cannot get it to switch.
Ideally within the unit test I would like to be able to mimic the switching of `StoreGraphs` within the `DualGraph` instance. I Initialised a DualGraph instance in an attempt to get switching to occur in test cases but still could not replicate the bug even while testing it on the current master which happens to have the bug still active in that branch.
If you have any ideas on this I'm all ears - If i cannot get it to replicate in a unit test should we remove the unit test as it is redundant? @arcturus2

Reply:
@aldebaran30701 i really like the idea of writing a unit test when fixing bugs as it helps detect it should it come back. I think you are mostly there with it and I'd like to help out to see if I can give some guidance to get the graph switching part to work. I'll get back to you.

Reply:
I've branched off master and added the unit test back in as to replicate the bug and work out how to get the graphs switching in the unit test. https://github.com/constellation-app/constellation/tree/layersSyncTest
I've tried accessing the instances from within the `LockingStoreGraph` but ran into many issues with private variables

I have just tried to get two writable graphs - both grabbed instance 1. Which is not what is needed.

Reply:
>I'd like to help out to see if I can give some guidance to get the graph switching part to work


@arcturus2 I think I'll take you up on that offer - Above is the branch I've been testing with.

Reply:
@aldebaran30701 please see PR #625 and let me know if this helps or not.